### ISSUE TITLE
PS7: Get-Error should report the fully-qualified error name

### ISSUE THREAD

**Issue by JustinGrote** [PowerShell/PowerShell#11106]
# Summary of the new feature/enhancement

Get-Error reports fullyqualifiederrorid without its namespace. This can make it difficult to find the fully qualified exception for use in catch statements, especially if running in Azure Functions or other environments with custom exceptions and you only have text logs to go off of. Since the major point of Get-Error is to address exceptions and perhaps catch them, the fully qualified type is important to do a specific catch, as my most common workflow when looking at errors is doing `$error[0].exception.gettype().fullname` as the very next step so I know what to use in a try/catch.

# Proposed technical implementation details (optional)

Add `FullyQualifiedErrorType` and `FullyQualifiedErrorParentType` properties that would display the full error type (what you would submit as a type to catch) as well as other error types that would catch this property, minus the obvious "system.exception" and "system.object" that all exceptions inherit from.

e.g.
FullyQualifiedErrorType:
```powershell
$error[0].exception.gettype()
```
FullyQualifiedErrorParentTypes:
```powershell
$currentType = $error[0].exception.getType()
while ($CurrentType) {
    $currentType = $currentType.BaseType
    if ($currentType.fullname -eq 'System.Exception') {break}
    $currentType
}
```

**Comment by mklement0** [comment]
Note that `.FullyQualifiedErrorId` is _not_ the same as the [type name of] the exception underlying the error record; when you create an error record, you can choose an error ID at your discretion; e.g.:

```
PS> [System.Management.Automation.ErrorRecord]::new([exception]::new(), 'MyErrorId', 'InvalidData', $null) | Get-Error

Exception             :
    Message : Exception of type 'System.Exception' was thrown.
    HResult : -2146233088
CategoryInfo          : InvalidData: (:) [], Exception
FullyQualifiedErrorId : MyErrorId
```

Note how the `FullyQualifiedErrorId:` line reflects the custom error ID passed to the error-record constructor - `MyErrorId` - as-is.

However, I agree that it _is_ desirable to have the full exception type name reflected in the output, by introducing _another output line_, e.g., `Type: `:

```powershell
Exception             :
    Type: System.Exception   # <<<< new line with full type name.
    Message : Exception of type 'System.Exception' was thrown.
    HResult : -2146233088
CategoryInfo          : InvalidData: (:) [], Exception
FullyQualifiedErrorId : MyErrorId
```

Ditto for `.InnerException`.

@SteveL-MSFT: Not sure if more work will be done before v7 GA; note that the VT sequences in `Get-Error`'s output are currently invariably emitted, even when you capture the output in a file.

**Comment by JustinGrote** [comment]
Right, my point was simply that there should be a property that at least
contains either the type definition of the exception or its string
representation, since 80% of the time, that's what I'm looking for.

**Comment by SteveL-MSFT** [comment]
I believe https://github.com/PowerShell/PowerShell/pull/11076 already resolves this?  Didn't make it in time for Preview.6, but should be there for RC.

**Comment by mklement0** [comment]
Great, @SteveL-MSFT; yes, that should resolve it.

However, can I suggest using just `Type:` rather than `ExceptionType:` as the label?
This would be consistent with the other labels; e.g., it's `Message:`, not `ExceptionMessage:`.

@JustinGrote, can you please close this issue?

**Comment by JustinGrote** [comment]
Totally missed the PR (there's a lot of them after all). Thanks @SteveL-MSFT. Meets my criteria.

Fixed by #11076

**Comment by mklement0** [comment]
@SteveL-MSFT, while you're working on #11076, can you look at these new issues?

* #11121
* #11122
* #11123
